compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -2514,24 +2514,16 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                                     tcx,
                                     infcx.fresh_substs_for_item(DUMMY_SP, impl_def_id),
                                 );
-                                // I guess we don't need to make a universe unless we need it,
-                                // but also we're on the error path, so it doesn't matter here.
-                                let universe = infcx.create_next_universe();
+                                let value = tcx.fold_regions(qself_ty, |_, _| tcx.lifetimes.re_erased);
```

Comment:
sidenote, we should rename this function to `fold_free_regions`, or even `map_free_regions`. Because currently it feels unclear whether this is supposed to modify bound regions

Reply:
"free" here is ambiguous because it also folds escaping bound vars, but yeah, I do think it needs a better name.